### PR TITLE
Add search box to the index page on Read the Docs 

### LINKED ISSUE #2481: Add search bar to main page of documentation
**Describe the flaw**
The start page of the documentation misses a search bar in the new theme.

**Suggest an improvement**
There should be a search bar on the documentation start page - ideally on the upper part of the page, since one often simply visits the page to search for a specific keyword (-> easy and instant access without scrolling or clicking would be very helpful).

**Supporting materials**
grafik

**Desktop/Environment**
system-independent issue

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by jessica-mitchell**
This PR resolves #2481.

It adds a search box using the material theme in the index.html template.

The only issue currently is that the font size seems to be smaller than other pages. @jens-br I think you have some experience with this, maybe you have an idea to get this identical to other pages?

See output
https://nest-test.readthedocs.io/en/add-searchbox/index.html

### CODE REVIEW

**Review by heplesser — APPROVED** [review]
Thanks!

**Comment by jens-br** [comment]
Hi @jessica-mitchell , it seems that the search bar gets scaled to 125%, but on the start page, it gets scaled back to 100% by the `normalize.css`. This results in the smaller height (38.4px vs 48px).
grafik

**Comment by jessica-mitchell** [comment]
@jens-br I noticed in my previous implementation, I broke the accordion feature on the index page. I found a different solution by doing the following
* in index.html 
    *  include  sphinx_material theme's header.html and add the application.css and material-icons.css (this one to make the search icon work)
 * in custom.css
     * add some styling to match colour and scale of the other pages (the default blue is used otherwise)

Still the rendering of the github repository icon and text is broken. I have not figured this one out yet and again would like to ask you if  you have an idea :)

Looks like this:
image

Should look like this:
image

**Comment by jens-br** [comment]
@jessica-mitchell Hi, sorry for the late reply, I am a bit busy at the moment. I looked at it, but I did not see the specific reason for that yet.

Was the removal of the div "md-source__facts" (screenshots below) intended?
grafik 

vs

 grafik

As `all: unset;` does not bring back the icon, I have the feeling that there might be some issue with `xlink:href="#__github"`.

Just a minor remark, aside the general problem: You seem to be using ``, but that is deprecated now. https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/xlink:href

File: doc/htmldoc/templates/index.html [hunk 1]
```diff
@@ -11,6 +11,7 @@
  <meta name="description" content="This is the documentation index for the NEST, a simulator for spiking neuronal networks.">
  <meta name="keywords" content="">
  <link rel="stylesheet" href="_static/stylesheets/application.css">
+ <link rel="stylesheet" href="_static/fonts/material-icons.css">
```
Comment (jens-br) [inline doc/htmldoc/templates/index.html:14]: Was that perhaps intended to be in `nest-simulator/doc/htmldoc/static/fonts` resp. `nest-simulator/doc/htmldoc/static/stylesheets/`? Or did I get that wrong?

**Comment by jessica-mitchell** [comment]
@jens-br The xlink is something that the sphinx-material theme uses; it also works on the other pages. And it seems the source__facts is somehow generated with application.js within the theme. Nothing here is explicit in our index.html.

**Comment by jessica-mitchell** [comment]
After some more digging around, there doesn't seem to be a straightforward way to get this repo icon to appear properly. I propose, for this PR, that we remove the repo from index.html, retaining the search box. I will create a new issue to try to resolve the issue of including sphinx-material theme in the index template.
How does that sound? @jens-br @steffengraber 

see: https://nest-test.readthedocs.io/en/add-searchbox/

**Comment by jens-br** [comment]
@jessica-mitchell Yes, that is a good idea. :+1: I will let you know if I find anything else.

**Comment by jessica-mitchell** [comment]
@jens-br ok great, could you then approve this pull request?

File: doc/htmldoc/templates/index.html (hunk 1 above)
Comment (jens-br) [reply doc/htmldoc/templates/index.html]: Ok :+1:

**Review by jens-br — APPROVED** [review]
LGTM :+1:

**Review by steffengraber — APPROVED** [review]
:+1:

**Pull request merged**